Add support for Deneb

realbigsean opened the pull request:
Resolves #3625

### Devnet V3 related notes [deprecated]

- install lighthouse on this branch by runnnig  `make`
  - Note: docker images for this branch available at `sigp/lighthouse:eip4844-amd64` (portable) and `sigp/lighthouse:eip4844-amd64-modern` (non-portable). Currently working on adding `aarch64` images, not available yet
- run:
  - `lighthouse bn --network eip4844 --execution-jwt <jwtsecret_path> --execution-endpoint http://localhost:8551  --trusted-setup-file <trusted_setup_file_path> --http`
- The `trustued-setup-file` to use: https://github.com/sigp/lighthouse/blob/eip4844/scripts/local_testnet/trusted_setup.txt 

### Devnet V4 related notes
- build the eip-4844 branch of this geth fork: https://github.com/mdehoog/go-ethereum
- `wget https://config.eip4844-devnet-4.ethpandaops.io/el/genesis.json`
- `geth init --datadir geth_data genesis.json`
- run geth
 ```
geth --ipcdisable --http --http.api="eth,web3,net,debug" --syncmode=full --bootnodes "enode://8488a57c323d5e5bef1c40c5cb2330cb0bd75d900afad5e7f50a47ecc1ae7683fb664f509908c6b978633b9af75788e943f26e9b64989504d189ae7b3278ede3@161.35.91.184:30303" --verbosity 4 --datadir geth_data --networkid 4844001004
```
- run lighthouse
```
lighthouse bn --purge-db --network eip4844 --execution-endpoint http://localhost:8551 --execution-jwt geth_data/geth/jwtsecret --debug-level trace --http --disable-optimistic-finalized-sync --libp2p-addresses "/ip4/64.225.74.231/tcp/9000/p2p/16Uiu2HAm4uoHpnkrfLncYRiKjWTtUfKUazH6afQgPtBuNpGDSJ7M"
```

### Local testnet instructions

GNU `grep` and GNU `sed` are required

- install lighthouse
  - `make`
- install lcli 
  - `make install-lcli`
- install this geth commit: https://github.com/mdehoog/go-ethereum/commit/2bd3a3cf9fa7d6a63326daf8c93b34ca5af86663
- update the geth_binary location here: https://github.com/sigp/lighthouse/blob/eip4844/scripts/local_testnet/vars.env#L1
- run `cd scripts/local_testnet`
- run `./start_local_testnet.sh -d debug genesis.json`
- all logs will be in `~/.lighthouse/local-testnet/testnet`
- stop the network with `./stop_local_testnet.sh`

Note: By default, this testnet starts at `Bellatrix`, forks to `Capella` at epoch 1 (slot 32) and forks to `EIP4844` at epoch 2 (slot 64). 

#### Update 12/28/22: "withdrawals" feature is no longer necessary as it's now enabled by default

jimmygchen commented:
~~@realbigsean maybe worth adding `json_rpc_snoop` installation to instructions as well:~~

~~- install `json_rpc_snoop`: https://github.com/ethDreamer/json_rpc_snoop~~

~~and your description:~~
~~> this is a proxy between lighthouse and geth that just prints out the json being sent back and forth...it's very useful for debugging CL <> EL compatibility~~

sorry, just realised you've already removed it :)

jimmygchen commented on Cargo.toml:
```diff
@@ -63,10 +63,12 @@ members = [
     "crypto/eth2_key_derivation",
     "crypto/eth2_keystore",
     "crypto/eth2_wallet",
+    "crypto/kzg",
```
^ this is added twice

realbigsean replied:
:+1:

emhane committed:
Boiler plate code for blobs pruning

emhane committed:
Prune blobs before data availability breakpoint

emhane committed:
Plug in pruning of blobs into app

emhane committed:
fixup! Prune blobs before data availability breakpoint

emhane committed:
fixup! Plug in pruning of blobs into app

emhane committed:
fixup! Plug in pruning of blobs into app

emhane committed:
Remove inaccurate guess for db index

emhane committed:
fixup! Prune blobs before data availability breakpoint

emhane committed:
Store blobs in separate freezer or historical state freezer

emhane committed:
fixup! Store blobs in separate freezer or historical state freezer

emhane committed:
Help user configure blobs freezer correctly between start ups

emhane committed:
fixup! Help user configure blobs freezer correctly between start ups

emhane committed:
Improve syntax

emhane committed:
Fix typo

Co-authored-by: realbigsean <seananderson33@GMAIL.com>

emhane committed:
Prune blobs on migration in addition to start-up

emhane committed:
Fix try_prune_blobs to use state root

emhane committed:
Improve syntax

Co-authored-by: Michael Sproul <micsproul@gmail.com>

emhane committed:
Save fetching state for blobs pruning

emhane committed:
Log info loaded from disk

emhane committed:
Delete blobs that conflict with finalization

emhane committed:
Store orphan block roots

emhane committed:
Prune blobs in bg after canonical head update

emhane committed:
Enable skipping blob pruning at each epoch

emhane committed:
fixup! Store orphan block roots

emhane committed:
Avoid repeteadly updating blob info for multiple head candidates

Madman600 committed:
Update checkpoint-sync.md (#3831)

Remove infura checkpoint sync instructions.


Co-authored-by: Adam Patacchiola <aep600@gmail.com>

santiagomed committed:
Return HTTP 404 rather than 405 (#3836)

Issue #3112

Add `Filter::recover` to the GET chain to handle rejections specifically as 404 NOT FOUND

Making a request to `http://localhost:5052/not_real` now returns the following:

```
{
    "code": 404,
    "message": "NOT_FOUND",
    "stacktraces": []
}
```

Co-authored-by: Paul Hauner <paul@paulhauner.com>

macladson committed:
Add CLI flag to specify the format of logs written to the logfile (#3839)

## Proposed Changes

Decouple the stdout and logfile formats by adding the `--logfile-format` CLI flag.
This behaves identically to the existing `--log-format` flag, but instead will only affect the logs written to the logfile.
The `--log-format` flag will no longer have any effect on the contents of the logfile.

## Additional Info
This avoids being a breaking change by causing `logfile-format` to default to the value of `--log-format` if it is not provided. 
This means that users who were previously relying on being able to use a JSON formatted logfile will be able to continue to use `--log-format JSON`. 

Users who want to use JSON on stdout and default logs in the logfile, will need to pass the following flags: `--log-format JSON --logfile-format DEFAULT`

emhane committed:
Convert epochs_per_blob_prune to Epoch once

emhane committed:
Fix epoch constructor misconception

emhane committed:
Simplify conceptual design

emhane committed:
fixup! Simplify conceptual design

infosecual committed:
add better err reporting UnableToOpenVotingKeystore (#3781)

## Issue Addressed

#3780 

## Proposed Changes

Add error reporting that notifies the node operator that the `voting_keystore_path` in their `validator_definitions.yml` file may be incorrect.

## Additional Info

There is more info in issue #3780 


Co-authored-by: Paul Hauner <paul@paulhauner.com>

GeemoCandama committed:
add logging for starting request  and receiving block (#3858)

## Issue Addressed

#3853 

## Proposed Changes

Added `INFO` level logs for requesting and receiving the unsigned block.

## Additional Info

Logging for successfully publishing the signed block is already there. And seemingly there is a log for when "We realize we are going to produce a block" in the `start_update_service`: `info!(log, "Block production service started");
`.  Is there anywhere else you'd like to see logging around this event?


Co-authored-by: GeemoCandama <104614073+GeemoCandama@users.noreply.github.com>

aliask committed:
Fix some dead links in markdown files (#3885)

## Issue Addressed

No issue has been raised for these broken links.

## Proposed Changes

Update links with the new URLs for the same document.

## Additional Info

~The link for the [Lighthouse Development Updates](https://eepurl.com/dh9Lvb/) mailing list is also broken, but I can't find the correct link.~


Co-authored-by: Paul Hauner <paul@paulhauner.com>

emhane committed:
fixup! Simplify conceptual design

emhane committed:
Remove IDE file

emhane committed:
Ignore IDE file

emhane committed:
Store blobs in correct db for atomic ops

emhane committed:
fixup! Store blobs in correct db for atomic ops

emhane committed:
Fix type bug

Co-authored-by: realbigsean <seananderson33@GMAIL.com>

emhane committed:
Help user choose blobs db

emhane committed:
Lint fix

emhane committed:
Add todos

emhane committed:
Clarify hybrid blob prune solution and fix error handling

emhane committed:
fixup! Clarify hybrid blob prune solution and fix error handling

emhane committed:
Fix typo

emhane committed:
Fix blobs store bug

emhane committed:
Run prune blobs on migrator thread

realbigsean committed:
fix/consolidate some error handling

realbigsean committed:
make explicity BlobsUnavailable error and handle it directly

realbigsean committed:
don't send errors when we *correctly* don't have blobs

realbigsean committed:
return a `BlobsUnavailable` error when the block root is a pre-4844 block

emhane committed:
Only store non-empty orphaned blobs

realbigsean committed:
update antithesis dockerfile (#3883)

Resolves https://github.com/sigp/lighthouse/issues/3879

Co-authored-by: realbigsean <sean@sigmaprime.io>

AgeManning committed:
Improve block delay metrics (#3894)

We recently ran a large-block experiment on the testnet and plan to do a further experiment on mainnet.

Although the metrics recovered from lighthouse nodes were quite useful, I think we could do with greater resolution in the block delay metrics and get some specific values for each block (currently these can be lost to large exponential histogram buckets). 

This PR increases the resolution of the block delay histogram buckets, but also introduces a new metric which records the last block delay. Depending on the polling resolution of the metric server, we can lose some block delay information, however it will always give us a specific value and we will not lose exact data based on poor resolution histogram buckets.

michaelsproul committed:
Switch allocator to jemalloc (#3697)

## Proposed Changes

Another `tree-states` motivated PR, this adds `jemalloc` as the default allocator, with an option to use the system allocator by compiling with `FEATURES="" make`.

- [x] Metrics
- [x] Test on Windows
- [x] Test on macOS
- [x] Test with `musl`
- [x] Metrics dashboard on `lighthouse-metrics` (https://github.com/sigp/lighthouse-metrics/pull/37)


Co-authored-by: Michael Sproul <micsproul@gmail.com>

emhane committed:
Fix typo

emhane committed:
Plug in running blob pruning in migrator, related bug fixes and add todos

emhane committed:
Allow user to set an epoch margin for pruning

emhane committed:
fixup! Allow user to set an epoch margin for pruning

emhane committed:
Improve error handling

antondlr committed:
fix multiarch docker builds (#3904)

## Issue Addressed

#3902 
Tested and confirmed working [here](https://github.com/antondlr/lighthouse/actions/runs/3970418322)

## Additional Info

buildx v0.10.0 added provenance attestations to images but they are packed in a way that's incompatible with `docker manifest`
https://github.com/docker/buildx/releases

realbigsean committed:
add new traits

michaelsproul committed:
Import BLS to execution changes before Capella (#3892)

* Import BLS to execution changes before Capella

* Test for BLS to execution change HTTP API

* Pack BLS to execution changes in LIFO order

* Remove unused var

* Clippy

realbigsean committed:
finish refactor

realbigsean committed:
remove unused imports

emhane committed:
fixup! Improve error handling

emhane committed:
Fix typo

emhane committed:
fixup! Improve error handling

emhane committed:
Fix network block_lookups test

emhane committed:
Debug tests

emhane committed:
fixup! Debug tests

realbigsean committed:
cleanup

emhane committed:
fixup! Debug tests

emhane committed:
Fix release test

emhane committed:
Add eip4844 fork to tests

realbigsean committed:
move available block comment

emhane committed:
Fix mismatched response bug

ensi321 committed:
Implement sync_committee_rewards API (per-validator reward) (#3903)

[#3661](https://github.com/sigp/lighthouse/issues/3661)

`/eth/v1/beacon/rewards/sync_committee/{block_id}`

```
{
  "execution_optimistic": false,
  "finalized": false,
  "data": [
    {
      "validator_index": "0",
      "reward": "2000"
    }
  ]
}
```

The issue contains the implementation of three per-validator reward APIs:
* `sync_committee_rewards`
* [`attestation_rewards`](https://github.com/sigp/lighthouse/pull/3822)
* `block_rewards`

This PR only implements the `sync_committe_rewards `.

The endpoints can be viewed in the Ethereum Beacon nodes API browser: [https://ethereum.github.io/beacon-APIs/?urls.primaryName=dev#/Rewards](https://ethereum.github.io/beacon-APIs/?urls.primaryName=dev#/Rewards)

The implementation of [consensus client reward APIs](https://github.com/eth-protocol-fellows/cohort-three/blob/master/projects/project-ideas.md#consensus-client-reward-apis) is part of the [EPF](https://github.com/eth-protocol-fellows/cohort-three).

Co-authored-by: navie <naviechan@gmail.com>
Co-authored-by: kevinbogner <kevbogner@gmail.com>

emhane committed:
Prune from highest data availability boundary

emhane committed:
fixup! Fix blobs store bug

emhane committed:
fixup! Prune from highest data availability boundary

emhane committed:
Reflect use of prune margin epochs at import

emhane committed:
Improve naming

emhane committed:
Atomically update blob info with pruned blobs

emhane committed:
Remove unused stuff

emhane committed:
Keep uniform size small keys

Co-authored-by: Michael Sproul <micsproul@gmail.com>

emhane committed:
Clarify wording

Co-authored-by: Michael Sproul <micsproul@gmail.com>

realbigsean committed:
simplify checking attester cache for block and blobs. use ResourceUnavailable according to the spec

emhane committed:
Add tests for blob pruning flags

emhane committed:
Correct comment

emhane committed:
Clarify comment

Co-authored-by: Michael Sproul <micsproul@gmail.com>

emhane committed:
Fix wording

Co-authored-by: Michael Sproul <micsproul@gmail.com>

divagant-martian committed:
only support 4844 rpc methods if on 4844

emhane committed:
fixup! Allow user to set an epoch margin for pruning

realbigsean committed:
Update beacon_node/lighthouse_network/src/rpc/methods.rs

realbigsean committed:
Update beacon_node/beacon_chain/src/beacon_chain.rs

realbigsean committed:
pass in data availability boundary to the get_blobs method

realbigsean committed:
review updates

realbigsean committed:
fix common issue across blocks by range and blobs by range

ethDreamer committed:
Use eth1_withdrawal_credentials in Test States (#3898)

* Use eth1_withdrawal_credential in Some Test States

* Update beacon_node/genesis/src/interop.rs

Co-authored-by: Michael Sproul <micsproul@gmail.com>

* Update beacon_node/genesis/src/interop.rs

Co-authored-by: Michael Sproul <micsproul@gmail.com>

* Increase validator sizes

* Pick next sync committee message

Co-authored-by: Michael Sproul <micsproul@gmail.com>
Co-authored-by: Paul Hauner <paul@paulhauner.com>

emhane committed:
Make implementation of BlobInfo more coder friendly

emhane committed:
Verify StoreConfig

GeemoCandama committed:
light client optimistic update reprocessing (#3799)

Currently there is a race between receiving blocks and receiving light client optimistic updates (in unstable), which results in processing errors. This is a continuation of PR #3693 and seeks to progress on issue #3651

Add the parent_root to ReprocessQueueMessage::BlockImported so we can remove blocks from queue when a block arrives that has the same parent root. We use the parent root as opposed to the block_root because the LightClientOptimisticUpdate does not contain the block_root.

If light_client_optimistic_update.attested_header.canonical_root() != head_block.message().parent_root() then we queue the update. Otherwise we process immediately.
michaelsproul came up with this idea.
The code was heavily based off of the attestation reprocessing.
I have not properly tested this to see if it works as intended.

michaelsproul committed:
Update sync rewards API for abstract exec payload

michaelsproul committed:
Fix the new BLS to execution change test

michaelsproul committed:
Update another test broken by the shuffling change

realbigsean committed:
Merge pull request #3910 from realbigsean/block-wrapper-refactor

Block wrapper refactor

realbigsean committed:
Merge pull request #3913 from divagant-martian/reject-pre-fork-rpc-upgrades

only support 4844 rpc methods if on 4844

realbigsean committed:
merge conflict resolution

realbigsean committed:
Merge pull request #3912 from realbigsean/sync-error-handling

Sync error handling

realbigsean committed:
update zip

emhane committed:
Fix weak_subjectivity_sync test

realbigsean committed:
Merge pull request #3915 from realbigsean/devnetv4-builtin-config

built in devnetv4 config eip4844

realbigsean committed:
remove unused import

emhane committed:
Fix range sync tests

emhane committed:
fixup! Fix block lookup debug tests

emhane committed:
Fix rebase conflicts

emhane committed:
Fix rebase conflicts

emhane committed:
Fix rebase conflicts

emhane committed:
Fixes after rebasing eip4844

emhane committed:
Fix clippy

emhane committed:
Fix release tests

emhane committed:
Change CI clippy

emhane committed:
fixup! Change CI clippy

realbigsean committed:
stub out tx root check, fix block hash calculation

realbigsean committed:
fix compile

realbigsean committed:
Merge pull request #3921 from realbigsean/cap-4844

Eip4844 capella rebase

realbigsean committed:
Merge branch 'eip4844' of https://github.com/sigp/lighthouse into fix-and-loosen-execution-block-decoding

realbigsean committed:
keep verification of payloads pre 4844

realbigsean committed:
Merge pull request #3920 from realbigsean/fix-and-loosen-execution-block-decoding

Fix and loosen execution block decoding

divagant-martian committed:
send stream terminators

realbigsean committed:
Update beacon_node/lighthouse_network/src/rpc/protocol.rs

realbigsean committed:
Merge pull request #3926 from divagant-martian/stream-timeout-blob-bug

send stream terminators

michaelsproul committed:
Fix regression in DB write atomicity

emhane committed:
Throw error when params don't match with previous run

emhane committed:
fixup! Throw error when params don't match with previous run

emhane committed:
Delete blobs along with block

emhane committed:
Complete making blocks and blobs db atomic

emhane committed:
fix rebase conflicts

emhane committed:
Fix rebase conflicts

emhane committed:
Fix rebase conflicts

emhane committed:
Fix rebase conflicts

realbigsean committed:
blob decoding

realbigsean committed:
blob tx decoding

realbigsean committed:
from to new

realbigsean committed:
improve error handling

realbigsean committed:
add back in 4844 tx consistencycheck during payload reconstruction

realbigsean committed:
variable list from -> new

realbigsean committed:
re-order methods

michaelsproul committed:
Unpin fixed-hash (#3917)

## Proposed Changes
Remove the `[patch]` for `fixed-hash`.

We pinned it years ago in #2710 to fix `arbitrary` support. Nowadays the 0.7 version of `fixed-hash` is only used by the `web3` crate and doesn't need `arbitrary`.

~~Blocked on #3916 but could be merged in the same Bors batch.~~

emhane committed:
fixup! Complete making blocks and blobs db atomic

emhane committed:
Fix rebase conflicts

emhane committed:
fixup! Complete making blocks and blobs db atomic

emhane committed:
Fix rebase conflict

emhane committed:
Fix rebase conflict

emhane committed:
Fix rebase conflict

emhane committed:
Fix conflicts rebasing eip4844

emhane committed:
Fix conflicts rebasing eip4844

emhane committed:
Fix conflicts rebasing eip4844

emhane committed:
Fix conflicts rebasing eip4844

emhane committed:
Fix conflicts rebasing eip4844

emhane committed:
Fix latest clippy lints

emhane committed:
fixup! Fix conflicts rebasing eip4844

emhane committed:
Improve use of whitespace

Co-authored-by: Michael Sproul <micsproul@gmail.com>

emhane committed:
Remove debug comment

Co-authored-by: Michael Sproul <micsproul@gmail.com>

emhane committed:
Reorder loading of db metadata from disk to allow for future changes to schema

emhane committed:
Removed unused code

emhane committed:
fixup! Improve use of whitespace

emhane committed:
fixup! Fix regression in DB write atomicity

realbigsean committed:
pr feedback and bugfixes

realbigsean committed:
Merge pull request #3941 from realbigsean/blob-decoding

Blob decoding

realbigsean committed:
Merge branch 'capella' of https://github.com/sigp/lighthouse into eip4844

# Conflicts:
#	beacon_node/beacon_chain/src/beacon_chain.rs
#	beacon_node/beacon_chain/src/block_verification.rs
#	beacon_node/beacon_chain/src/test_utils.rs
#	beacon_node/execution_layer/src/engine_api.rs
#	beacon_node/execution_layer/src/engine_api/http.rs
#	beacon_node/execution_layer/src/lib.rs
#	beacon_node/execution_layer/src/test_utils/handle_rpc.rs
#	beacon_node/http_api/src/lib.rs
#	beacon_node/http_api/tests/fork_tests.rs
#	beacon_node/network/src/beacon_processor/mod.rs
#	beacon_node/network/src/beacon_processor/work_reprocessing_queue.rs
#	beacon_node/network/src/beacon_processor/worker/sync_methods.rs
#	beacon_node/operation_pool/src/bls_to_execution_changes.rs
#	beacon_node/operation_pool/src/lib.rs
#	beacon_node/operation_pool/src/persistence.rs
#	consensus/serde_utils/src/u256_hex_be_opt.rs
#	testing/antithesis/Dockerfile.libvoidstar

realbigsean committed:
fix compilation issues after merge

divagant-martian committed:
self rate limiting

michaelsproul committed:
Update dependencies (#3946)

Resolves the cargo-audit failure caused by https://rustsec.org/advisories/RUSTSEC-2023-0010.

I also removed the ignore for `RUSTSEC-2020-0159` as we are no longer using a vulnerable version of `chrono`. We still need the other ignore for `time 0.1` because we depend on it via `sloggers -> chrono -> time 0.1`.

realbigsean committed:
use hex decode instead of parse

emhane committed:
Fix conflicts rebasing eip4844

emhane committed:
fixup! Fix latest clippy lints

realbigsean committed:
check byte length when converting to uint256 and hash256 from bytes. Add comments

realbigsean committed:
remove clone of access lists

realbigsean committed:
fix lints

realbigsean committed:
Merge pull request #3852 from emhane/prune_blobs

Prune blobs

emhane committed:
Fix conflicts from rebasing eip4844

emhane committed:
fixup! Help user choose blobs db

emhane committed:
Remove unused dependencies

realbigsean committed:
pr feedback

emhane committed:
fixup! Update dependencies (#3946)

emhane committed:
fixup! Add tests for blob pruning flags

emhane committed:
Fix lint for prune blobs pr

emhane committed:
Run cargo update

emhane committed:
fixup! Add tests for blob pruning flags

realbigsean committed:
Merge pull request #3869 from emhane/blobs_freezer

Blobs freezer

realbigsean committed:
Merge pull request #3951 from realbigsean/fix-blob-tx-ssz

Encode blob transactions as signed blob transactions

realbigsean committed:
merge upstream, add back `get_blobs` logic

realbigsean committed:
allow unused

realbigsean committed:
complete match for `has_context_bytes`

realbigsean committed:
remove unused structs

realbigsean committed:
make batch size check compile time panic

realbigsean committed:
make batch size check compile time panic

realbigsean committed:
Revert "make batch size check compile time panic"

This reverts commit 68f2484efcccd3a51fd1f92356cf95671d0c9eb1.

realbigsean committed:
fix a couple new lints

emhane committed:
Merge pull request #2 from realbigsean/sean-debug-ci

Debug CI Sean's PR feedback

emhane committed:
Fix merge conflicts with eip4844

emhane committed:
Remove unused error

emhane committed:
Disable use of system time in tests

emhane committed:
Comment out tests that use system time

emhane committed:
fixup! Disable use of system time in tests

realbigsean committed:
add v3 to capabilities

michaelsproul committed:
Upgrade sqlite3

emhane committed:
Fix lint

emhane committed:
Better fix for debug tests

emhane committed:
Remove unused generic

realbigsean committed:
Merge pull request #3905 from emhane/beacon_chain_tests

Debug CI

realbigsean committed:
merge with upstream

realbigsean committed:
merge self limiter

realbigsean committed:
merge self limiter

realbigsean committed:
fix compilation after merge

realbigsean committed:
bump recursion limit

divagant-martian committed:
Merge branch 'unstable' into off-4844

divagant-martian committed:
solve couple of merge conflicts

divagant-martian committed:
Revert "Use consensus-spec-tests `v1.3.0-rc.3` (#4021)"

This reverts commit caa6190d4a468ef74d647f668ac04e4accf293d5.

divagant-martian committed:
define BELLATRIX_FORK_EPOCH in scripts/tests/vars.env with same value as ALTAIR_FORK_EPOCH

divagant-martian committed:
define CAPELLA_FORK_EPOCH in scripts/tests/vars.env with same value as ALTAIR_FORK_EPOCH

divagant-martian committed:
define missing vars

divagant-martian committed:
Revert "Clean capella (#4019)"

This reverts commit 047c7544e3ea4d3277a86602a68d32ca3228c87f.

divagant-martian committed:
Merge branch 'unstable-up-to-clean-capella' into eip4844

divagant-martian committed:
Merge branch 'unstable' into eip4844

divagant-martian committed:
fix gitignore

divagant-martian committed:
add missing eip4844 core topics placeholder

divagant-martian committed:
Merge branch 'unstable' into eip4844

divagant-martian committed:
new lints

divagant-martian committed:
bump up recursion limits by a lot

divagant-martian committed:
more lints

divagant-martian committed:
bump up recursion limits in the sim bin

divagant-martian committed:
Merge branch 'unstable' into eip4844

divagant-martian committed:
fix merge mistake

divagant-martian committed:
Merge branch 'unstable' into eip4844

divagant-martian committed:
Merge commit '65a5eb829264cb279ed66814c961991ae3a0a04b' into eip4844

divagant-martian committed:
Merge branch 'unstable' into eip4844

divagant-martian committed:
revert change to ef_tests

divagant-martian committed:
Merge commit '036b797b2c1831352f937356576b3c78c65220ad' into eip4844

divagant-martian committed:
Merge commit 'c5383e393acee152e92641ce4699d05913953e70' into eip4844

divagant-martian committed:
Merge commit '1e029ce5384e911390a513e2d1885532f34a8b2b' into eip4844

divagant-martian committed:
Merge branch 'unstable' into eip4844

divagant-martian committed:
fix merge conflict

divagant-martian committed:
Merge branch 'unstable' into eip4844

pawanjay176 committed:
Merge branch 'unstable' into eip4844

pawanjay176 committed:
Fix compiler issues

realbigsean commented:
Closing in favor of https://github.com/sigp/lighthouse/pull/4054
